config/routes.rb

```diff
@@ -3,4 +3,10 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  resources :dishes, only: [:show]
+  post "/dishes/:id", to: "dishes#create"
```

Comment:
As mentioned in the comment in your controller, a more RESTful path and route here would be:
`post "/dishes/:dish_id/ingredients, to: "dish_ingredients#create"`

Reply:
You absolutely do not need to use advanced routing in an IC unless you want to (it's not required!) but if you wanted to make this route using `resources`:
```
resources :dishes, only: :show do
    resources :ingredients, only: :create, controller: "dish_ingredients"
end
```